tests/unit/test_signers.py

```diff
@@ -250,6 +250,63 @@ def test_presigned_url_throws_unsupported_signature_error(self):
             self.signer.generate_presigned_url({})
 
 
+class TestCloudfrontSigner(unittest.TestCase):
+    def setUp(self):
+        self.signer = CloudFrontSigner("MY_KEY_ID", lambda message: b'signed')
+
+    def test_b64encode(self):
+        self.assertEqual(b'dGVzdA__', self.signer._url_b64encode(b'test'))
+
+    def test_build_canned_policy(self):
+        policy = self.signer.build_policy('foo', datetime.datetime(2016, 1, 1))
+        expected = (
+            b'{"Statement":[{"Resource":"foo",'
+            b'"Condition":{"DateLessThan":{"AWS:EpochTime":1451606400}}}]}')
+        self.assertEqual(policy, expected)
+
+    def test_build_custom_policy(self):
+        policy = self.signer.build_policy(
+            'foo', datetime.datetime(2016, 1, 1),
+            date_greater_than=datetime.datetime(2015, 12, 1),
+            ip_address='12.34.56.78/9')
+        expected = (
```

Comment:
I like your idea and started working on it.

And, because the order of all keys in the policy dictionary is significant, I tried OrderedDict first. But it turns out the error diff of OrderedDict shows up as a long string, which defeats the purpose.

So I end up using a normal dict (which has better error diff by default), yet I still keep the current string-based diff to ensure the correct order.